.azure-pipelines.yml

```diff
@@ -42,6 +42,18 @@ jobs:
   - bash: |
       pre-commit run --all-files --show-diff-on-failure
     displayName: 'Pre-commit hooks check'
+  - bash: |
+      if ([[ $1 = "refs/heads/master" ]] || [[ $1 = "refs/tags/"* ]])
+      then exit 0; fi
+      for FILENAME in newsfragments/*
```

Comment:
To avoid the for loop:
`[[ $(git --no-pager diff master --name-only -- newsfragments/ | grep -v README.rst | wc -l) -eq 0 ]]`

Reply:
Added clearer comment